extensions/ql-vscode/src/variant-analysis/run-remote-query.ts

```diff
@@ -242,19 +279,36 @@ function isFileSystemRoot(dir: string): boolean {
   return pathObj.root === dir && pathObj.base === "";
 }
 
-async function createRemoteQueriesTempDirectory() {
-  const remoteQueryDir = await dir({
+interface RemoteQueryTempDir {
+  remoteQueryDir: DirectoryResult;
+  queryPackDir: string;
+  compiledPackDir: string;
+  bundleFile: string;
+}
+
+async function createRemoteQueriesTempDirectory(): Promise<RemoteQueryTempDir> {
+  const shortRemoteQueryDir = await dir({
     dir: tmpDir.name,
     unsafeCleanup: true,
   });
+  // Expand 8.3 filenames here to work around a CLI bug where `codeql pack bundle` produces an empty
```

Comment:
How big of a problem do you expect this to be in practice? My research says they're a feature from pre-windows 95 and now only supported for compatibility with old systems. We don't consider these 8.3 filenames anywhere else, but should we? How likely are we to encounter them with users?

I think it's impressive that we're handling them and I don't want to discount the work you've done here, but the code in `short-paths.ts` is quite complex so I just want to be sure that the maintenance burden is going to be worth it compared to how many users it will unblock. Do you have any more insights or data on this?

Reply:
Sorry I didn't read the PR description well enough. I see you've explained that you hit this on the actions runners and why it's happening. In that case I see you're point that it's only really a case for CI and not for user machines. I still think some unit tests would be helpful, unless they're really hard to write. But I see the reason why this is implemented.